tests/contrib/elasticsearch/test_elasticsearch.py

```diff
@@ -225,20 +225,51 @@ def test_patch_unpatch(self):
         assert spans, spans
         assert len(spans) == 1
 
-    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
-    def test_user_specified_service(self):
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc", DD_TRACE_SPAN_ATTRIBUTE_SCHEMA="v0"))
+    def test_user_specified_service_v0(self):
         """
-        When a user specifies a service for the app
+        v0: When a user specifies a service for the app
             The elasticsearch integration should not use it.
         """
         assert config.service == "mysvc"
 
+        self.es.indices.create(index=self.ES_INDEX, ignore=400)
+        spans = self.get_spans()
+        self.reset()
+        assert len(spans) == 1
+        assert spans[0].service != "mysvc"
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc", DD_TRACE_SPAN_ATTRIBUTE_SCHEMA="v1"))
+    def test_user_specified_service_v1(self):
+        """
+        v1: When a user specifies a service for the app
+            The elasticsearch integration should use it.
+        """
+        assert config.service == "mysvc"
+
+        self.es.indices.create(index=self.ES_INDEX, ignore=400)
+        Pin(service="es", tracer=self.tracer).onto(self.es.transport)
+        spans = self.get_spans()
+        self.reset()
+        assert len(spans) == 1
+        assert spans[0].service == "mysvc"
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_TRACE_SPAN_ATTRIBUTE_SCHEMA="v0"))
+    def test_unspecified_service_v0(self):
+        self.es.indices.create(index=self.ES_INDEX, ignore=400)
+        spans = self.get_spans()
+        self.reset()
+        assert len(spans) == 1
+        assert spans[0].service == "elasticsearch"
+
+    @TracerTestCase.run_in_subprocess(env_overrides=dict(DD_TRACE_SPAN_ATTRIBUTE_SCHEMA="v1"))
+    def test_user_specified_service_v1(self):
         self.es.indices.create(index=self.ES_INDEX, ignore=400)
         Pin(service="es", tracer=self.tracer).onto(self.es.transport)
         spans = self.get_spans()
         self.reset()
         assert len(spans) == 1
-        assert spans[0].service != "es"
+        assert spans[0].service == "unnamed-python-service"
```

Comment:
nit: Probably stick to importing `DEFAULT_SPAN_SERVICE_NAME` and using that?

Reply:
will do!